### PR TITLE
Fmcomms5 rebased

### PR DESCRIPTION
**Pull request by SrikanthPagadarai**
Rebased FMCOMMS5 commits.

### CODE REVIEW

File: +adi/+FMComms5/Rx.m [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+            % methods
+            setupLibad9361(obj);
+            obj.iioDevPHYChipB = calllib('libiio', 'iio_context_find_device',obj.iioCtx,'ad9361-phy-B');
+            % Do writes directly to hardware without using set methods.
+            % This is required sine Simulink support doesn't support
+            % modification to nontunable variables at SetupImpl
+            
+            % Gains
+            obj.setAttributeRAW('voltage0','gain_control_mode',obj.GainControlModeChannel0,false);
+            if obj.channelCount>2
```
Comment (tfcollins) [inline +adi/+FMComms5/Rx.m:243]: Might not want to use channelCount here. If EnabledChannels == [3,4] the gain modes will not be updated

File: +adi/+FMComms5/Tx.m [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+    %
+    %   <a href="https://www.analog.com/en/design-center/evaluation-hardware-and-software/evaluation-boards-kits/eval-ad-fmcomms5-ebz.html">Product Page</a>
+    
+    properties (Nontunable)
+        %DataSource Data Source
+        %   Data source, specified as one of the following:
+        %   'DMA' — Specify the host as the source of the data.
+        %   'DDS' — Specify the DDS on the radio hardware as the source
+        %   of the data. In this case, each channel has two additive tones.
+        DataSourceChipB = 'DMA';
```
Comment (tfcollins) [inline +adi/+FMComms5/Tx.m:17]: Does this have any effect?

File: +adi/+FMComms5/Tx.m [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+                flag = flag || strcmpi(prop,'CenterFrequencyChipB');
+            end
+        end
+    end
+    
+    %% API Functions
+    methods (Hidden, Access = protected)
+        function ToggleDDSChipB(obj,value)
+            chanPtr = getChan(obj,obj.iioDevChipB,'altvoltage0',true);
+            iio_channel_attr_write_bool(obj,chanPtr,'raw',value);
```
Comment (tfcollins) [inline +adi/+FMComms5/Tx.m:269]: Is this method required?

File: +adi/+FMComms5/Tx.m [hunk 4]
```diff
@@ -0,0 +1,356 @@
...
+                    obj.setAttributeLongLong('voltage0','sampling_frequency',obj.SamplingRate,true,4,obj.iioDevPHYChipB);
+                    obj.setAttributeLongLong('voltage0','rf_bandwidth',obj.RFBandwidthChipB,strcmp(obj.Type,'Tx'),[],obj.iioDevPHYChipB);
+                end
+            else
+                writeFilterFile(obj);
+                writeFilterFileFMComms5ChipB(obj);
+            end            
+            
+            obj.setAttributeLongLong('voltage0','hardwaregain',obj.AttenuationChannel0,true);
+            if (obj.channelCount>2)
```
Comment (tfcollins) [inline +adi/+FMComms5/Tx.m:338]: Same as RX, this will lead to gains not set cases like [3,4]==EnabledChannels

Comment (SrikanthPagadarai) [reply +adi/+FMComms5/Tx.m]: Addressed by the commit.

File: +adi/+FMComms5/Tx.m (hunk 3 above)
Comment (SrikanthPagadarai) [reply +adi/+FMComms5/Tx.m:253]: Addressed by the commit.

File: +adi/+FMComms5/Tx.m (hunk 2 above)
Comment (SrikanthPagadarai) [reply +adi/+FMComms5/Tx.m]: Addressed by the commit.

File: +adi/+FMComms5/Rx.m (hunk 1 above)
Comment (SrikanthPagadarai) [reply +adi/+FMComms5/Rx.m]: Addressed by the commit.

File: +adi/+FMComms5/Rx.m [hunk 5]
```diff
@@ -0,0 +1,313 @@
...
+            
+            % Gains
+            if (any(obj.EnabledChannels == 1))
+                obj.setAttributeRAW('voltage0','gain_control_mode',obj.GainControlModeChannel0,false);
+            end
+            if (any(obj.EnabledChannels == 2))
+                obj.setAttributeRAW('voltage1','gain_control_mode',obj.GainControlModeChannel1,false);
+            end
+            if (any(obj.EnabledChannels == 3))
+                obj.setAttributeRAW('voltage0','gain_control_mode',obj.GainControlModeChannel0,false,obj.iioDevPHYChipB);
```
Comment (tfcollins) [inline +adi/+FMComms5/Rx.m:249]: GainControlModeChannel0 -> GainControlModeChannel0ChipB

File: +adi/+FMComms5/Rx.m [hunk 6]
```diff
@@ -0,0 +1,313 @@
...
+            end
+            if (any(obj.EnabledChannels == 4))
+                obj.setAttributeRAW('voltage1','gain_control_mode',obj.GainControlModeChannel1,false,obj.iioDevPHYChipB);
```
Comment (tfcollins) [inline +adi/+FMComms5/Rx.m:252]: GainControlModeChannel1 -> GainControlModeChannel1ChipB

File: +adi/+FMComms5/Tx.m [hunk 7]
```diff
@@ -0,0 +1,337 @@
...
+                    obj.setAttributeLongLong('voltage0','sampling_frequency',obj.SamplingRate,true,4,obj.iioDevPHYChipB);
+                    obj.setAttributeLongLong('voltage0','rf_bandwidth',obj.RFBandwidthChipB,strcmp(obj.Type,'Tx'),[],obj.iioDevPHYChipB);
+                end
+            else
+                writeFilterFile(obj);
+                writeFilterFileFMComms5ChipB(obj);
+            end            
+            
+            if (any(obj.EnabledChannels == 1))
+                obj.setAttributeRAW('voltage0','gain_control_mode',obj.GainControlModeChannel0,true);
```
Comment (tfcollins) [inline +adi/+FMComms5/Tx.m:317]: Need to change these 4 from gain control to gains

File: +adi/+FMComms5/Rx.m (hunk 5 above)
Comment (SrikanthPagadarai) [reply +adi/+FMComms5/Rx.m]: Fixed.

File: +adi/+FMComms5/Rx.m (hunk 6 above)
Comment (SrikanthPagadarai) [reply +adi/+FMComms5/Rx.m]: Fixed.

File: +adi/+FMComms5/Tx.m (hunk 7 above)
Comment (SrikanthPagadarai) [reply +adi/+FMComms5/Tx.m]: You are right. Fixed them.

**Pull request merged**